src/ai/backend/manager/models/image.py

```diff
@@ -892,6 +893,33 @@ async def calculate_final_permission(self, rbac_obj: ImageRow) -> frozenset[Imag
         return frozenset(permissions)
 
 
+@dataclass
+class ImageAccessCriteria:
+    user_id: UUID
+    allowed_registries: set[str]
+
+    def is_accessible_image(self, image: ImageRow) -> bool:
+        """
+        Check if the image is accessible by the user.
+        """
+        if image.registry not in self.allowed_registries:
+            return False
+        if image.labels.get(
+            "ai.backend.customized-image.owner"
+        ) is not None and not image.is_customized_by(self.user_id):
+            return False
+        return True
+
+    def is_accessible_customized_image(self, image: ImageRow) -> bool:
+        """
+        Check if the image is a customized image accessible by the user.
+        """
+        if image.labels.get("ai.backend.customized-image.owner") is None:
+            return False
+
+        return image.is_customized_by(self.user_id) and self.is_accessible_image(image)
```

Comment:
It seems like all you just need to do is create the `customized()`, `is_owner(user_id)` methods on the ImageRow and `is_accessible(image_row)` on the ImageAccessCriteria.